metadata/utils.go

```diff
@@ -23,18 +23,17 @@ import (
 )
 
 // GetMetadataProperty returns a property from the metadata map, with support for case-insensitive keys and aliases.
-func GetMetadataProperty(props map[string]string, keys ...string) (val string, ok bool) {
+func GetMetadataProperty(props map[string]string, keys ...string) (key string, val string, ok bool) {
```

Comment:
Please create a separate method `GetMetadataPropertyWithMatchedKey` (or something else) rather than modifying the exiting API. This way, we don't have to change the API in dozens of places!

Reply:
Once `GetMetadataPropertyWithMatchedKey` is implemented, `GetMetadataProperty` can change to:

```go
func GetMetadataProperty(props map[string]string, keys ...string) (val string, ok bool) {
_, val, ok = GetMetadataPropertyWithMatchedKey(props, keys...)
return val, ok
}